src/libraries/System.DirectoryServices/ref/System.DirectoryServices.manual.cs

```diff
@@ -4,9 +4,15 @@
 // Changes to this file must follow the https://aka.ms/api-review process.
 // ------------------------------------------------------------------------------
 
+using System.DirectoryServices.Design;
+
 namespace System.DirectoryServices
 {
     [System.ComponentModel.TypeConverter(typeof(DirectoryEntryConverter))]
     public partial class DirectoryEntry { }
-    internal sealed class DirectoryEntryConverter { }
 }
+
+namespace System.DirectoryServices.Design
+{
+    internal sealed class DirectoryEntryConverter { }
```

Comment:
@ericstj do you remember why the internal type is part of / exposed in the contract?

Reply:
The XAML compiler records the TypeConverter at design time (even if it is internal). https://github.com/dotnet/runtime/issues/27528

It would do this for any type, [even internal types](https://github.com/dotnet/wpf/blob/a7ab0e9192c10f0ca6c3ad4b351d74e6c4494f1a/src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/Markup/BamlMapTable.cs#L596).  All these type converters (internal as well) were exposed in the .NETFramework assemblies and this behavior was something WPF depended upon.  There were cases we saw that relied on these internal converters.  We brought it back for any type from .NETFramework that was exposed in .NET Core assemblies.

This behavior is specifically why we brought back TypeConverterAttributes into reference assemblies - it's why those became promoted to public surface area (including their internal targets).